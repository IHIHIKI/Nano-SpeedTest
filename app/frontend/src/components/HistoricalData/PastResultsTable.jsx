import React from 'react';
import PropTypes from 'prop-types';
import PastResultsTableRow from './PastResultsTableRow';
import uuid from 'uuid';
import 'styles/PastResults.css';

class PastResultsTable extends React.Component {
    constructor(props) {
        super(props);
        this.state = {
        };
    }

    render() {
        const {tableData} = this.props;

        return (
              <div className="your-transations-wrapper">
                  <div className="past__transactions_title__wrapper center-horizontally max-width">
                     <div className="past-transations-title">Past users transactions</div>
                     <a className="past__transactions__link" href="/Stats">See More</a>
                  </div>
          <table className="transactions-table max-width">
          <tbody>
             <tr className="hidden-on-mobile">
                <th className="border_bottom">Status</th>
                <th className="border_bottom">Origin</th>
                <th className="border_bottom">Destination</th>
                <th className="border_bottom">Amount</th>
                <th className="border_bottom">Elapsed Time</th>
                <th className="border_bottom">Sending Block</th>
             </tr>
                    { tableData &&
                        tableData.sort((a,b) => b.endSendTimestamp - a.endSendTimestamp)
                        .map((transactionData) => {
                            return <PastResultsTableRow key={uuid(transactionData.id)} {...transactionData}/>;
                        })
                    }
                    </tbody>
                  </table>
               </div>
        );
    }
}

PastResultsTable.propTypes = {
    tableData: PropTypes.array
};

export default PastResultsTable;