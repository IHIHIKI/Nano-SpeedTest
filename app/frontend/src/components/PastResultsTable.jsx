import React from 'react';
import PropTypes from 'prop-types';
import PastResultsTableRow from './PastResultsTableRow';
import uuid from 'uuid';

const PastResultsTable = ({tableData}) => {
    return (
        <div className='nano-container'>
            <table className='table'>
                <thead>
                <tr>
                    <th scope='col'>Id</th>
                    <th scope='col'>Origin</th>
                    <th scope='col'>Destination</th>
                    <th scope='col'>Amount</th>
                    <th scope='col'>Elapsed Time</th>
                    <th scope='col'>Date</th>
                </tr>
                </thead>
                <tbody>
                {
                    tableData.reverse().map((transactionData, index) => {
                        return <PastResultsTableRow key={uuid(transactionData.id)} {...transactionData}/>;
                    })
                }
                </tbody>
            </table>
        </div>
    );
};

PastResultsTable.propTypes = {
    tableData: PropTypes.array
};

export default PastResultsTable;