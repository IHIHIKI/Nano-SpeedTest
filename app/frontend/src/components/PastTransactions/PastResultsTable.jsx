import React from 'react';
import PropTypes from 'prop-types';
import PastResultsTableRow from './PastResultsTableRow';
import uuid from 'uuid';

const maxItems = 25;

class PastResultsTable extends React.Component {
    state = {
        startIndex: 0
    }
    render() {
        const {tableData} = this.props;
        const {startIndex} = this.state;
        // const endIndex = (startIndex + maxItems) < tableData.length ? startIndex + maxItems : 
        return (
            <div className='nano-container'>
                <h2 className='map-header text-left'>Past Transactions</h2>
                <table className='table'>
                    <thead>
                    <tr>
                        <th scope='col'>Id</th>
                        <th scope='col'>Origin</th>
                        <th scope='col'>Destination</th>
                        <th scope='col'>Amount</th>
                        <th scope='col'>Elapsed Time</th>
                        <th scope='col'>Date</th>
                    </tr>
                    </thead>
                    <tbody>
                    {
                        tableData.reverse().slice(startIndex, startIndex + maxItems).map((transactionData, index) => {
                            return <PastResultsTableRow key={uuid(transactionData.id)} {...transactionData}/>;
                        })
                    }
                    </tbody>
                </table>
            </div>
        );
    }
};

PastResultsTable.propTypes = {
    tableData: PropTypes.array
};

export default PastResultsTable;