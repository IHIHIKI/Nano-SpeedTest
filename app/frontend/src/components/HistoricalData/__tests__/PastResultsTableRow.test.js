import React from 'react';
import renderer from 'react-test-renderer';
import { shallow } from 'enzyme';
import PastResultsTable from '../PastResultsTable';

const props = {
    index: 0,
    id: 123,
    origin: {},
    destination: {},
    tableData: []
};

it('shallow renders without crashing', () => {
    shallow(<PastResultsTable key={123} {...props}/>);
});

it(`renders snapshot correctly`, () => {
    const tree = renderer.create(
        <PastResultsTable key={123} {...props}/>
    ).toJSON();
    expect(tree).toMatchSnapshot();
});
