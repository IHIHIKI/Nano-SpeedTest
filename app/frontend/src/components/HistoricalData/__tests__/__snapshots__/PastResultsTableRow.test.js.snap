// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders snapshot correctly 1`] = `
<div
  className="nano-container"
>
  <h2
    className="map-header text-left"
  >
    Past Transactions
  </h2>
  <table
    className="table"
  >
    <thead>
      <tr>
        <th
          scope="col"
        >
          Origin
        </th>
        <th
          scope="col"
        >
          Destination
        </th>
        <th
          scope="col"
        >
          Amount
        </th>
        <th
          scope="col"
        >
          Elapsed Time
        </th>
        <th
          scope="col"
        >
          Date
        </th>
        <th
          className="block-col"
          scope="col"
        >
          Sending Block
        </th>
        <th
          className="block-col"
          scope="col"
        >
          Receiving Block
        </th>
      </tr>
    </thead>
    <tbody />
  </table>
  <div
    className="row"
  >
    <div
      className="col-sm-auto"
    >
      Showing 
      1
       to 
      0
       of 
      0
       transactions.
    </div>
    <div
      className="col-sm-auto"
    >
      <nav
        aria-label="Table navigation"
      >
        <ul
          className="pagination"
        >
          <li
            className="page-item disabled"
          >
            <button
              className="page-link"
              onClick={[Function]}
            >
              Prev
            </button>
          </li>
          <li
            className="page-item disabled"
          >
            <button
              className="page-link"
              onClick={[Function]}
            >
              Next
            </button>
          </li>
        </ul>
      </nav>
    </div>
  </div>
</div>
`;
